src/components/patch/PatchContainer.tsx

```diff
@@ -51,41 +57,64 @@ export class PatchContainer extends React.Component<Props, State> {
             />
           </Paper>
         </div>
-        <Patches />
+        <InfiniteScroll hasMore={this.state.hasMore} loadMore={this.loadPatches} initialLoad={false}>
+          <Patches />
+        </InfiniteScroll>
       </div>
     );
   }
 
-  private loadPatches() {
-    this.props.client.getPatches((err, resp, body) => {
-      const versions = ConvertToPatches(JSON.stringify(resp.body)).VersionsMap
-      this.setState({ 
-        versions: versions, 
-        visible: versions, 
-      });
-    }, this.props.client.username);
+  private loadPatches = () => {
+    if (this.state.hasMore) {
+      this.props.client.getPatches((err, resp, body) => {
+        const newPatches = ConvertToPatches(resp.body).VersionsMap;
+        if (Object.keys(newPatches).length === 0) {
+          this.setState({ 
+            hasMore: false 
+          });
+          return;
+        }
+        const sortedPatches = this.state.allPatches;
```

Comment:
Do we need to do this? I would think that the API returns patches in reverse chronological order, so that we only need to append the new results

Reply:
The API does return them in order, but with the oldest patch first (we want the newest patch first). I already needed a compare by date function elsewhere in the code so I just used it here as well to order everything.